ffi/diplomat/src/properties_maps.rs

```diff
@@ -0,0 +1,77 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+#[diplomat::bridge]
+pub mod ffi {
+    use alloc::boxed::Box;
+    use icu_properties::{
+        maps::{self, CodePointMapResult},
+        provider::{UnicodePropertyMapV1, UnicodePropertyMapV1Marker},
+        Script,
+    };
+    use icu_provider::prelude::DataPayload;
+
+    use crate::{provider::ffi::ICU4XDataProvider, provider::ffi::ICU4XStaticDataProvider};
+
+    #[diplomat::opaque]
+    /// An ICU4X Unicode Set Property object, capable of querying whether a code point is contained in a set based on a Unicode property. For properties whose values fit into 16 bits.
+    /// See [the Rust docs](https://unicode-org.github.io/icu4x-docs/doc/icu_properties/index.html) for more information.
+    pub struct ICU4XCodePointMapData16(DataPayload<'static, UnicodePropertyMapV1Marker<u16>>);
+
+    pub struct ICU4XCodePointMapData16Response {
+        /// The [`ICU4XCodePointMapData16`], if creation was successful.
+        pub data: Option<Box<ICU4XCodePointMapData16>>,
+        /// Whether creating the [`ICU4XCodePointMapData16`] was successful.
+        pub success: bool,
+    }
+
+    impl ICU4XCodePointMapData16 {
+        /// Gets a map for Unicode property Script from a [`ICU4XDataProvider`].
+        /// See [the Rust docs](https://unicode-org.github.io/icu4x-docs/doc/icu_properties/maps/fn.get_script.html) for more information.
+        pub fn try_get_script(provider: &ICU4XDataProvider) -> ICU4XCodePointMapData16Response {
+            let provider = provider.0.as_ref();
+            Self::prepare_result_from_script(maps::get_script(provider))
+        }
+
+        /// Gets a map for Unicode property Script from a [`ICU4XStaticDataProvider`].
+        /// See [the Rust docs](https://unicode-org.github.io/icu4x-docs/doc/icu_properties/maps/fn.get_script.html) for more information.
+        pub fn try_get_script_from_static(
+            provider: &ICU4XStaticDataProvider,
+        ) -> ICU4XCodePointMapData16Response {
+            let provider = provider.0.as_ref();
+            Self::prepare_result_from_script(maps::get_script(provider))
+        }
+
+        fn prepare_result_from_script(
+            result: CodePointMapResult<'static, Script>,
+        ) -> ICU4XCodePointMapData16Response {
+            match result {
+                Ok(data) => {
+                    let data: DataPayload<UnicodePropertyMapV1Marker<u16>> = data
+                        .try_map_project_with_capture((), |data_struct, _, _| {
+                            match data_struct.code_point_trie.try_into_converted::<u16>() {
```

Comment:
suggestion(nb): you could replace the match block with a `.map`

```rust
match data_struct.code_point_trie.try_into_converted::<u16>()
    .map(|code_point_trie| UnicodePropertyMapV1 { code_point_trie })
```

I'm not sure if that's more readable. It's a minor change.